### PR TITLE
Delete order front

### PR DESCRIPTION
**Pull request by engshorouq**
- fix a bug on a popup

### CODE REVIEW

File: client/src/Components/CommonComponent/Table/Table.jsx [hunk 1]
```diff
@@ -11,13 +11,13 @@ import "./style.css";
...
   state = {
     pageSize: "10",
-    singleCustomer: {
-      editVisibilty: false,
-      deleteVisibility: false,
-      viewVisibility: false,
-      id: '',
-      information: null
-    },
+    // singleCustomer: {
```
Comment (hshahwan) [inline client/src/Components/CommonComponent/Table/Table.jsx]: delete  comment lines

File: client/src/Components/CommonComponent/Table/Table.jsx [hunk 2]
```diff
@@ -264,54 +265,33 @@ class TableCmponent extends Component {
...
-                  />
                   <Divider type="vertical" />
-                    <Icon onClick={this.handleClick("singleCustomer", "deleteVisibility", record.key, record)}
+                    <Icon onClick = {this.props.viewValues("singleCustomer", "deleteVisibility", record.key, record)}
                       style={{
                         fontSize: "1.2rem",
                         color: "rgba(0, 0, 0, 0.65)"
                       }}
                       type="delete"
                       className={record.key}
```
Comment (hshahwan) [inline client/src/Components/CommonComponent/Table/Table.jsx]: why you used  ``` className={record.key}```?
If not necessary, delete it

File: client/src/Components/CommonComponent/Table/Table.jsx [hunk 3]
```diff
@@ -476,12 +456,12 @@ class TableCmponent extends Component {
 
 TableCmponent.propTypes = {
   columns: PropTypes.array.isRequired,
-  viewPopup: PropTypes.func.isRequired,
-  editPopup: PropTypes.func.isRequired,
-  deletePopup: PropTypes.func.isRequired,
-  viewHtml: PropTypes.string.isRequired,
-  editHtml: PropTypes.string.isRequired,
-  deleteHtml: PropTypes.string.isRequired,
+  // viewPopup: PropTypes.func.isRequired,
```
Comment (hshahwan) [inline client/src/Components/CommonComponent/Table/Table.jsx]: return prop-types validation for popups.

File: client/src/Components/Layouts/SingleCustomer/index.jsx [hunk 4]
```diff
@@ -76,32 +83,46 @@ class Profile extends Component {
...
     });
-    this.setState({tableInfo: table});
-  }
-  handleClick = () => {
-    this.setState({ visible: true });
+    this.setState({ tableInfo: table });
+  };
+  deleteRow = id => {
+    this.setState(prev => {
+      return { tableInfo: prev.tableInfo.filter(data => data.key !== id) };
```
Comment (tomduggan85) [inline client/src/Components/Layouts/SingleCustomer/index.jsx]: @engshorouq You don't need to use the updater function here inside setState, you could just access this.state directly:
`this.setState({ tableInfo: this.state.tableInfo.filter( .....`

**Pull request merged**